### ISSUE TITLE
Exceptions while importing Keras models

### ISSUE THREAD

**Issue by maltyyev** [deeplearning4j/deeplearning4j#3173]
Hi,
I'm trying to import these two models from Keras:
`HIDDEN_SIZE = 128
    LAYERS = 1

    model = Sequential()
    model.add(LSTM(HIDDEN_SIZE, input_shape=((input_MAXLEN, alphabet_length))))
    model.add(RepeatVector(label_MAXLEN))
    for i in range(LAYERS):
        model.add(LSTM(HIDDEN_SIZE, return_sequences=True))
    model.add(TimeDistributed(Dense(alphabet_length)))
    model.add(Activation('softmax'))
    model.compile(loss='categorical_crossentropy',
                  optimizer='adam',
                  metrics=['accuracy'])`
I get the exception: `Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.UnsupportedKerasConfigurationException: Unknown keras weight initializer orthogonal. Please file an issue at http://github.com/deeplearning4j/deeplearning4j/issues.
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.mapWeightInitialization(KerasLayer.java:637)
	at org.deeplearning4j.nn.modelimport.keras.layers.KerasLstm.getRecurrentWeightInitFromConfig(KerasLstm.java:310)
	at org.deeplearning4j.nn.modelimport.keras.layers.KerasLstm.<init>(KerasLstm.java:82)
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.getKerasLayerFromConfig(KerasLayer.java:214)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.helperPrepareLayers(KerasModel.java:190)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:99)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:56)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel$ModelBuilder.buildSequential(KerasModel.java:581)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:156)
	at ImportingFromKeras.main(ImportingFromKeras.java:17)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
`
And the second model:
`inputs = Input(shape=(maxlen, vocab_size), name='input', dtype='float32')

    conv = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[0],
                         border_mode='valid', activation='relu',
                         input_shape=(maxlen, vocab_size))(inputs)
    conv = MaxPooling1D(pool_length=3)(conv)

    conv1 = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[1],
                          border_mode='valid', activation='relu')(conv)
    conv1 = MaxPooling1D(pool_length=3)(conv1)

    conv2 = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[2],
                          border_mode='valid', activation='relu')(conv1)

    conv3 = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[3],
                          border_mode='valid', activation='relu')(conv2)

    conv4 = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[4],
                          border_mode='valid', activation='relu')(conv3)

    conv5 = Convolution1D(nb_filter=nb_filter, filter_length=filter_kernels[5],
                          border_mode='valid', activation='relu')(conv4)
    conv5 = MaxPooling1D(pool_length=3)(conv5)
    conv5 = Flatten()(conv5)

    z = Dropout(0.5)(Dense(dense_outputs, activation='relu')(conv5))
    z = Dropout(0.5)(Dense(dense_outputs, activation='relu')(z))

    pred = Dense(cat_output, activation='softmax', name='output')(z)

    model = Model(input=inputs, output=pred)

    sgd = SGD(lr=0.01, momentum=0.9)
    model.compile(loss='categorical_crossentropy', optimizer=sgd,
                  metrics=['accuracy'])`
The exception:
`Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.UnsupportedKerasConfigurationException: Unsupported keras layer type Convolution1D. Please file an issue at http://github.com/deeplearning4j/deeplearning4j/issues.
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.getKerasLayerFromConfig(KerasLayer.java:254)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.helperPrepareLayers(KerasModel.java:190)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:164)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:97)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel$ModelBuilder.buildModel(KerasModel.java:576)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:122)
	at ImportingFromKeras.main(ImportingFromKeras.java:18)`
I believe that at this point (dl4j version 0.8.0) 1D Convolutional layers are supported so I have no idea why second exception rises.
Thank you.

**Comment by turambar** [comment]
@maltyyev both of these are known.

Re: 1D convolutions -- DL4J now supports [1D convolutional layers](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution1DLayer.java) (1D pooling, too), but we haven't finished support for importing them from Keras. Still a few kinks to work out, but it's coming.

Re: orthogonal weight initializations, that is not yet supported.

^^^^^ @AlexDBlack take note of the request for orthogonal weight initialization.

**Comment by turambar** [comment]
Since this is redundant with other issues, I'm going to close it.

- Convolution1D request: https://github.com/deeplearning4j/deeplearning4j/issues/3101
- Orthogonal weight initialization: https://github.com/deeplearning4j/deeplearning4j/issues/2748
